File: successfulPairsOfSpells&Potions.py
from typing import List
import bisect

class Solution:
    def successfulPairs(self, spells: List[int], potions: List[int], success: int) -> List[int]:
        potions.sort()
        m = len(potions)
        ans = []
        for s in spells:
            # minimal potion needed: ceil(success / s)
            need = (success + s - 1) // s
            idx = bisect.bisect_left(potions, need)   # first potion >= need
            ans.append(m - idx)
        return ans

File: successfulPairsOfSpells&Potions.go
package main

func successfulPairsBrute(spells []int, potions []int, success int64) []int {
	n := len(spells)
	ans := make([]int, n)
	for i, s := range spells {
		var cnt int
		ss := int64(s)
		for _, p := range potions {
			if ss*int64(p) >= success {
				cnt++
			}
		}
		ans[i] = cnt
	}
	return ans
}
